File: food_coon_app/lib/models/menu_category.dart
class MenuCategory {
  final int id;
  final int restaurantId;
  final String name;

  MenuCategory({
    required this.id,
    required this.restaurantId,
    required this.name,
  });

  factory MenuCategory.fromJson(Map<String, dynamic> json) {
    return MenuCategory(
      id: json['id'] as int,
      restaurantId: json['restaurantId'] as int,
      name: json['name'] as String,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'restaurantId': restaurantId,
        'name': name,
      };
}

File: food_coon_rn/src/models/MenuCategory.ts
export interface MenuCategory {
  id: number;
  restaurantId: number;
  name: string;
  sortOrder: number;
}

export function menuCategoryFromJson(json: Record<string, any>): MenuCategory {
  return {
    id: json.id,
    restaurantId: json.restaurantId,
    name: json.name,
    sortOrder: json.sortOrder ?? 0,
  };
}
